api/pkg/cli/cmd/install/install.go

```diff
@@ -134,13 +134,23 @@ func (opts *options) run() error {
 		}
 	}
 
-	installer := installer.New(opts.cs)
-	opts.resource, err = installer.Install(manifest, opts.from, opts.cs.Namespace())
-	if err != nil {
-		return opts.errors(err)
+	out := opts.cli.Stream().Out
+
+	resourceInstaller := installer.New(opts.cs)
+
+	resource, errors := resourceInstaller.Install(manifest, opts.from, opts.cs.Namespace())
+	opts.resource = resource
+
+	if len(errors) != 0 {
+		// fetch the tekton pipelines version
+		pipelinesVersion := resourceInstaller.TektonPipelinesVersion()
```

Comment:
you can save pipeline version in `resourceInstaller` after the first call and then use it.. this will save a call to cluster

Reply:
here https://github.com/vinamra28/hub/blob/728e303f71a7dbcfecc3fdf031e3099d5bdc11a9/api/pkg/cli/installer/config.go#L28